x-pack/platform/plugins/shared/alerting/common/routes/rule_template/apis/find/schemas/v1.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { stringOrStringArraySchema } from '../../../../../schemas';
+
+export const findRuleTemplatesRequestQuerySchema = schema.object({
```

Comment:
nit: We cannot paginate more than 10K templates. It is a limitation of ES. I am wondering if we should add a validation where we throw if `Math.max(page, page * per_page) > 10K`

Reply:
ok i'll add a limitation for now. how do other 'find' APIs handle this? do they implement it properly using `search_after`? is there no helper for this?